### PR TITLE
[AEA-562] Feature/config new variables

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

AEA config optional parameters added: timeout, execution_timeout, max_reactions, search_service_address (for OEF).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -98,6 +100,7 @@ def __init__(
             self.decision_maker.preferences,
             self.decision_maker.goal_pursuit_readiness,
             self.task_manager,
+            search_service_address,
```
Comment (DavidMinarsch) [inline aea/aea.py:103]: I am sorry, but I think we need to revert this. The AEA should not have to be aware of this value. We need to pass it to the agent context in a different manner. I will make a proposal ...

File: aea/aea_builder.py [hunk 2]
```diff
@@ -243,11 +245,56 @@ def __init__(self, with_default_packages: bool = True):
         self._default_connection = DEFAULT_CONNECTION
         self._context_namespace = {}  # type: Dict[str, Any]
 
+        self._timeout: Optional[float] = None
+        self._execution_timeout: Optional[float] = None
+        self._max_reactions: Optional[int] = None
+        self._search_service_address: Optional[str] = None
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:251]: see above

File: aea/configurations/schemas/aea-config_schema.json [hunk 3]
```diff
@@ -98,6 +98,21 @@
...
+      "type": ["number", "null"],
+      "minimum": 0
+    },
+    "max_reactions": {
+      "type": ["integer", "null"],
+      "minimum": 1
+    },
+    "search_service_address": {
+    	"type": ["string", "null"]
     }
```
Comment (DavidMinarsch) [inline aea/configurations/schemas/aea-config_schema.json:116]: please move all the type definitions to `definitions`

Comment (solarw) [reply aea/configurations/schemas/aea-config_schema.json]: done

File: aea/aea_builder.py (hunk 2 above)
Comment (solarw) [reply aea/aea_builder.py]: removed

File: aea/aea.py (hunk 1 above)
Comment (solarw) [reply aea/aea.py]: removed

**Comment by solarw** [comment]
> LGTM, we should discuss the one issue I mentioned
@DavidMinarsch 
merged with latest dev

File: aea/aea.py [hunk 4]
```diff
@@ -73,6 +73,7 @@ def __init__(
         :param exeution_timeout: amount of time to limit single act/handle to execute.
         :param is_debug: if True, run the agent in debug mode (does not connect the multiplexer).
         :param max_reactions: the processing rate of envelopes per tick (i.e. single loop).
+        :param search_service_address: OEF address to use.
```
Comment (DavidMinarsch) [inline aea/aea.py:76]:
```suggestion
```

File: aea/context/base.py [hunk 5]
```diff
@@ -150,11 +146,6 @@ def task_manager(self) -> TaskManager:
         """Get the task manager."""
         return self._task_manager
```
Comment (DavidMinarsch) [inline aea/context/base.py:148]:
```suggestion
        return self._task_manager

    @property	
    def search_service_address(self) -> Address:	
        """Get the address of the search service."""	
        return self._search_service_address

```

File: aea/context/base.py [hunk 6]
```diff
@@ -73,10 +73,6 @@ def __init__(
         self._preferences = preferences
         self._goal_pursuit_readiness = goal_pursuit_readiness
         self._task_manager = task_manager
```
Comment (DavidMinarsch) [inline aea/context/base.py:75]:
```suggestion
        self._task_manager = task_manager
        self._search_service_address = (	
            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml	
        )	
```

File: aea/skills/base.py [hunk 7]
```diff
@@ -45,7 +45,7 @@
 from aea.crypto.ledger_apis import LedgerApis
 from aea.decision_maker.base import GoalPursuitReadiness, OwnershipState, Preferences
 from aea.helpers.base import add_modules_to_sys_modules, load_all_modules, load_module
-from aea.mail.base import Address, OutBox
+from aea.mail.base import OutBox
```
Comment (DavidMinarsch) [inline aea/skills/base.py:48]:
```suggestion
from aea.mail.base import Address, OutBox
```

File: aea/skills/base.py [hunk 8]
```diff
@@ -190,11 +190,6 @@ def ledger_apis(self) -> LedgerApis:
         """Get ledger APIs."""
         return self._get_agent_context().ledger_apis
```
Comment (DavidMinarsch) [inline aea/skills/base.py:192]:
```suggestion
        return self._get_agent_context().ledger_apis

    @property	
    def search_service_address(self) -> Address:	
        """Get the address of the search service."""	
        return self._get_agent_context().search_service_address	

```

File: aea/skills/base.py [hunk 9]
```diff
@@ -190,10 +190,10 @@ def ledger_apis(self) -> LedgerApis:
...
 
-    @property
-    def search_service_address(self) -> Address:
-        """Get the address of the search service."""
-        return self._get_agent_context().search_service_address
+    @property	
+    def search_service_address(self) -> Address:	
+        """Get the address of the search service."""	
+        return self._get_agent_context().search_service_address
```
Comment (DavidMinarsch) [inline aea/skills/base.py:196]:
```suggestion
    @property
    def search_service_address(self) -> Address:
        """Get the address of the search service."""
        return self._get_agent_context().search_service_address
```

File: aea/context/base.py [hunk 10]
```diff
@@ -150,9 +149,9 @@ def task_manager(self) -> TaskManager:
         """Get the task manager."""
         return self._task_manager
 
-    @property
-    def search_service_address(self) -> Address:
-        """Get the address of the search service."""
+    @property	
+    def search_service_address(self) -> Address:	
+        """Get the address of the search service."""
```
Comment (DavidMinarsch) [inline aea/context/base.py:154]:
```suggestion
    @property
    def search_service_address(self) -> Address:
        """Get the address of the search service."""
```

File: aea/context/base.py [hunk 11]
```diff
@@ -73,10 +73,9 @@ def __init__(
         self._preferences = preferences
         self._goal_pursuit_readiness = goal_pursuit_readiness
         self._task_manager = task_manager
-        self._search_service_address = (
-            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml
-        )
-
+        self._search_service_address = (	
+            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml	
+        )
```
Comment (DavidMinarsch) [inline aea/context/base.py:78]:
```suggestion
        )

```

File: aea/context/base.py [hunk 12]
```diff
@@ -73,8 +73,8 @@ def __init__(
         self._preferences = preferences
         self._goal_pursuit_readiness = goal_pursuit_readiness
         self._task_manager = task_manager
-        self._search_service_address = (
-            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml
+        self._search_service_address = (	
+            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml
```
Comment (DavidMinarsch) [inline aea/context/base.py:77]:
```suggestion
        self._search_service_address = (
            DEFAULT_OEF  # TODO: make this configurable via aea-config.yaml
```

**Pull request merged**